appserver/web/weld-integration/src/main/java/org/glassfish/weld/JaxRSJsonContextResolver.java

```diff
@@ -117,4 +120,13 @@ public Jsonb getContext(Class<?> type) {
             }
         });
     }
+
+    private void instantiateResolverClasses() {
+        if (!existingResolverClasses.isEmpty()) {
+            existingResolverClasses.stream().map(injectionManager::getInstance)
+                    .filter(Objects::nonNull)
+                    .forEach(existingResolvers::add);
+        }
+        existingResolverClasses.clear();
```

Comment:
@lprimak It is unclear why you want to clear this *outside* of the `!isEmpty()` case.

Reply:
fixed